presto-spi/src/main/java/com/facebook/presto/spi/function/StandardFunctionResolution.java

```diff
@@ -68,6 +68,14 @@ public interface StandardFunctionResolution
 
     FunctionHandle countFunction(Type valueType);
 
+    boolean supportsArbitraryFunction();
+
+    boolean isArbitraryFunction(FunctionHandle functionHandle);
+
+    FunctionHandle arbitraryFunction();
+
+    FunctionHandle arbitraryFunction(Type valueType);
+
```

Comment:
This is a change in SPI which in general we want to minimize the the change, we may not need it.
Can you check how we are doing it here https://github.com/prestodb/presto/blob/00a30bd74c717690e2742fcfb7ecb965d015de97/presto-main-base/src/main/java/com/facebook/presto/sql/planner/iterative/rule/CombineApproxPercentileFunctions.java#L203 ?

Reply:
While in general we want to minimize SPI changes, this one is harmless because approx_percentile has long been a part of the canon of Presto functions.